docs/tutorials/tutorials.md

```diff
@@ -0,0 +1,382 @@
+      </doc>
+    </group>
+    ```
+
+In the above example the group `transmission_correction` is an optional group.
+
+#### Keyword `unit`
+The `unit` keyword is used to define the NeXus-compliant unit categories.
+
+**`unit` in YAML**
+
+=== "YAML"
+    ```yaml
+    detector_voltage(NX_FLOAT):
+      unit: NX_VOLTAGE
+      doc: |
+        Voltage applied to detector.
+    ```
+=== "XML"
+    ```xml
+    <field name="detector_voltage" type="NX_FLOAT">
+      <unit>NX_VOLTAGE</unit>
+      <doc>Voltage applied to detector.</doc>
+    </field>
+    ```
+In the above example, the `detector_voltage` field is defined as a `NX_FLOAT` type with a unit of `NX_VOLTAGE`, indicating that the values stored in this field are measured in volts or millivolts or other voltage units.
+
+#### Keyword `dimensions`
+
+The `dimensions` term  describes the multidimensional nature of the data, specifying its rank, dimensional indices, and corresponding length of the rank. The attribute `rank` defines the dimension of the data set. To elucidate each dimension, we use two other keywords: `dim` and `dim_parameters`. The `dim` keyword comprises an array of arrays, the nested array encapsulates values for `index` and `value` (NeXus keywords) pairs. Each array within the `dim` array corresponds to a specific dimension of the multidimensional data. For example, for 2D particle motion, the `dim` array may be represented as `[[0, nx], [1, ny]]`, each member indicating the axis index and axis name. The keyword `dim_parameters` contains further information of each dimension such as `doc`, `ref`, etc. It is important to note that each array corresponds to a keyword within `dim_parameters` must have the same length as the value of the `rank` keyword.
+
+
+**`dimensions` in YAML**
+
+=== "YAML"
+    ```yaml
+    # 2D particle motion
+    dimensions:
+      rank: 2
+      dim: [[0, nx], [1, ny]]
+      dim_parameters:
+          doc: ["Position of particle on x-axis.","Position of particle on y-axis."]
+    ```
+    The `dimensions` can also be written in shorter form
+    **Dimensions in YAML (shorter form)**
+=== "YAML (shorter form)"
+    ```yaml
+    # 2D particle motion
+    dimensions:
+      rank: 2
+      dim: (nx, ny)
+    ```
+=== "XML"
+    ```xml
+    <dimensions rank="2">
+      <dim index="0" value="nx">
+        <doc>Position of particle on x-axis.</doc>
+      </dim>
+      <dim index="1" value="ny">
+        <doc>Position of particle on y-axis.</doc>
+      </dim>
+    </dimensions>
+    ```
+=== "XML (shorter form)"
+    ```xml
+    <dimensions rank="2">
+      <dim index="0" value="nx"/>
+      <dim index="1" value="ny"/>
+    </dimensions>
+    ```
+
+#### Keyword `enumeration`
+List of strings which are considered as a set of predefined values for fields or attributes. Individual items of the enumeration may also hold a `doc` keyword to provide a description of the item.
```

Comment:
for `field` and attribute` hyphenation like in the other paragraphs that explain keywords for the sake of consistency